src/common/inference/service.rs

```diff
@@ -138,12 +139,6 @@ impl InferenceService {
     pub fn init_global(config: InferenceConfig) -> Result<(), StorageError> {
         let mut inference_service = INFERENCE_SERVICE.write();
 
-        if config.token.is_none() {
```

Comment:
So now we rely solely on this code for the token extraction?

```rust
  let token = inference_token
            .0
            .unwrap_or_else(|| self.config.token.clone().unwrap_or_default());
```

Does the inference service fail with a 4xx error with an empty token?

Reply:
yes, the inference service fail with a 4xx error with an empty token. it's a change for upcoming release.